docs/nsm_api.md

```diff
@@ -6,11 +6,193 @@ Based on the discussions on NSM IRC channel on September 11th/12th and in the is
 
 
 
-## List of NSM components
+## List of Network Service Mesh Components
+
+### Network Service Mesh Components in the Abstract
+
+Current Network Service Mesh is highly focused on Network Service Mesh within the context of a Kubernetes Cluster.  Most of this document will work within that context.
+Network Service Mesh concepts are highly genericizable.
+
+![Network Service Mesh Abstract Concepts](./images/NSM%20Diagrams%20for%20Arch%20Docs.png)
+<dl>
+    <dt>Network Service (NS)</dt>
+    <dd>
+        A Network Service is the abstract representation of the various behaviors to be provided to a Network Service Client via an L2/L3 connection.  It can include:
+        <ul>
+            <li>Connectivity to Isolated Resources</li>
+            <li>Protection for Threats</li>
+            <li>Guaranteed Bandwidth/Latency</li>
+            <li>Load Balancing</li>
+            <li>Proxying</li>
+        </ul>
+        A Network Service has a name and a payload.  The payload defines the type of L2/L3 payload (Ethernet/IPv4/IPv6/MPLS/etc) the Network Service accepts.
```

Comment:
I am not sure if `payload` is the right term here. As example L2 domain can be used to carry different types of payloads. Even if payload is Ethernet, Ethernet frames could carry other types of payloads.  I think something more like `connectivity_type` is a better match here.

Reply:
right, but there is ambiguity. Why not to just to give a name which would not trigger questions like mine?